hera_cal/lstbin.py

```diff
@@ -190,13 +253,16 @@ def lst_bin(data_list, lst_list, flags_list=None, dlst=None, begin_lst=None, lst
     else:
         # keep all lst_grid bins and fill empty ones with unity data and mark as flagged
         for index in range(len(lst_grid)):
-            if index in all_lst_indices:
-                # skip if index already in data
+            if index in all_lst_indices and baselines_same_across_nights:
```

Comment:
`all_lst_indices` represents the union of LSTs across all nights and baselines. We can have situations where `index` is in `all_lst_indices` but it may not be present for a particular baseline which occurs on a subset of the nights being averaged. This is not a problem if the baselines sets are all the same on all the nights which is why I added the additional `bls_same_across_nights` condition.